src/windows/mod.rs

```diff
@@ -0,0 +1,407 @@
+//! Windows-specific implementation of the `SharedLibrary` trait.
+
+use crate::Segment as SegmentTrait;
+use crate::SharedLibrary as SharedLibraryTrait;
+use crate::{Bias, IterationControl, SharedLibraryId, Svma};
+
+use std::ffi::{CStr, OsStr, OsString};
+use std::fmt;
+use std::marker::PhantomData;
+use std::mem;
+use std::os::windows::ffi::OsStringExt;
+use std::ptr;
+use std::slice;
+use std::usize;
+
+use winapi::ctypes::c_char;
+use winapi::shared::guiddef::GUID;
+use winapi::shared::minwindef::{HMODULE, MAX_PATH};
+use winapi::um::libloaderapi::{FreeLibrary, LoadLibraryExW, LOAD_LIBRARY_AS_DATAFILE};
+use winapi::um::memoryapi::VirtualQuery;
+use winapi::um::processthreadsapi::GetCurrentProcess;
+use winapi::um::psapi::{
+    EnumProcessModules, GetModuleFileNameExW, GetModuleInformation, MODULEINFO,
+};
+use winapi::um::winnt::{
+    IMAGE_DEBUG_DIRECTORY, IMAGE_DEBUG_TYPE_CODEVIEW, IMAGE_DIRECTORY_ENTRY_DEBUG,
+    IMAGE_DOS_HEADER, IMAGE_DOS_SIGNATURE, IMAGE_NT_HEADERS, IMAGE_NT_SIGNATURE,
+    IMAGE_SCN_CNT_CODE, IMAGE_SECTION_HEADER, MEMORY_BASIC_INFORMATION, MEM_COMMIT,
+};
+
+const CV_SIGNATURE: u32 = 0x5344_5352;
+
+/// An unsupported segment
+pub struct Segment<'a> {
+    section: &'a IMAGE_SECTION_HEADER,
+}
+
+impl<'a> fmt::Debug for Segment<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("Segment")
+            .field("name", &self.name())
+            .field("is_code", &self.is_code())
+            .finish()
+    }
+}
+
+impl<'a> SegmentTrait for Segment<'a> {
+    type SharedLibrary = SharedLibrary<'a>;
+
+    #[inline]
+    fn name(&self) -> &str {
+        std::str::from_utf8(&self.section.Name)
+            .unwrap_or("")
+            .trim_end_matches('\0')
+    }
+
+    fn is_code(&self) -> bool {
+        (self.section.Characteristics & IMAGE_SCN_CNT_CODE) != 0
+    }
+
+    #[inline]
+    fn stated_virtual_memory_address(&self) -> Svma {
+        Svma(self.section.VirtualAddress as usize)
+    }
+
+    #[inline]
+    fn len(&self) -> usize {
+        *unsafe { self.section.Misc.VirtualSize() } as usize
+    }
+}
+
+/// An iterator over PE sections.
+pub struct SegmentIter<'a> {
+    sections: &'a [IMAGE_SECTION_HEADER],
+}
+
+impl<'a> fmt::Debug for SegmentIter<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("SegmentIter").finish()
+    }
+}
+
+impl<'a> Iterator for SegmentIter<'a> {
+    type Item = Segment<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.sections.is_empty() {
+            None
+        } else {
+            let section = &self.sections[0];
+            self.sections = &self.sections[1..];
+            Some(Segment { section })
+        }
+    }
+}
+
+#[repr(C)]
+struct CodeViewRecord70 {
+    signature: u32,
+    pdb_signature: GUID,
```

Comment:
```suggestion
    pdb_signature: [u8; 16],
```

So that pointer casts aren't needed below.